api/app/signals/apps/email_integrations/tests/test_rules.py

```diff
@@ -0,0 +1,220 @@
+# Copyright (C) 2021 - 2022 Gemeente Amsterdam
+from django.test import TestCase
+from factory.fuzzy import FuzzyText
+
+from signals.apps.email_integrations.rules import (
+    SignalCreatedRule,
+    SignalHandledRule,
+    SignalOptionalRule,
+    SignalReactionRequestReceivedRule,
+    SignalReactionRequestRule,
+    SignalReopenedRule,
+    SignalScheduledRule
+)
+from signals.apps.questionnaires.app_settings import NO_REACTION_RECEIVED_TEXT
+from signals.apps.signals import workflow
+from signals.apps.signals.factories import SignalFactory, StatusFactory
+
+
+class RuleTestMixin:
+    rule = None
+    state = None
+
+    def test_happy_flow(self):
+        status_text = FuzzyText(length=200) if self.state == workflow.REACTIE_GEVRAAGD else FuzzyText(length=400)
+        signal = SignalFactory.create(status__state=self.state, status__text=status_text,
+                                      reporter__email='test@example.com')
+        self.assertTrue(self.rule(signal))
+
+    def test_signal_not_happy_flow(self):
+        signal = SignalFactory.create(status__state=workflow.BEHANDELING, reporter__email='test@example.com')
+        self.assertFalse(self.rule(signal))
+
+    def test_anonymous_reporter(self):
+        status_text = FuzzyText(length=200) if self.state == workflow.REACTIE_GEVRAAGD else FuzzyText(length=400)
+
+        signal = SignalFactory.create(status__state=self.state, status__text=status_text, reporter__email='')
+        self.assertFalse(self.rule(signal))
+
+        signal = SignalFactory.create(status__state=self.state, status__text=status_text, reporter__email=None)
+        self.assertFalse(self.rule(signal))
+
+    def test_apply_for_parent_signals(self):
+        status_text = FuzzyText(length=200) if self.state == workflow.REACTIE_GEVRAAGD else FuzzyText(length=400)
+
+        parent_signal = SignalFactory.create(status__state=self.state, status__text=status_text,
+                                             reporter__email='test@example.com')
+        SignalFactory.create(status__state=self.state, status__text=status_text, reporter__email='test@example.com',
+                             parent=parent_signal)
+
+        self.assertTrue(self.rule(parent_signal))
+
+    def test_do_not_apply_for_child_signals(self):
+        status_text = FuzzyText(length=200) if self.state == workflow.REACTIE_GEVRAAGD else FuzzyText(length=400)
+
+        parent_signal = SignalFactory.create(status__state=self.state, status__text=status_text,
+                                             reporter__email='test@example.com')
+        child_signal = SignalFactory.create(status__state=self.state, status__text=status_text,
+                                            reporter__email='test@example.com', parent=parent_signal)
+
+        self.assertFalse(self.rule(child_signal))
+
+    def test_invalid_states(self):
+        states = list(map(lambda x: x[0] and x[0] == self.state, workflow.STATUS_CHOICES))
```

Comment:
This creates a list of `True` and `False` values, not a list of status codes (Status.state strings). I think your intention here is more something like:

```
states = [s[0] for s in workflow.STATUS_CHOICES if s != self.state
```

As an aside `SignalFactory.create(status__state=True, reporter__email='test@example.com')` works.

Reply:
Good catch. Fixed it. And strange that it accepts `status_state=True` 🤔